src/RouteJs.AspNet/router.js

```diff
@@ -160,7 +160,7 @@
 			///<param name="processedParams">Array of parameters that have already been processed</param>
 			///<returns type="String">URL with parameters merged in, or null if not all parameters were specified</returns>
 			
-			var finalUrl = this.route.url;
+			var finalUrl = this._isLowerCase() ? this.route.url.toLowerCase() : this.route.url;
```

Comment:
If you pass a settings object in the constructor and ensure it always has a `lowerCaseUrls` property, you could just do `this.settings.lowerCaseUrls ? this.route.url.toLowerCase() : this.route.url` and avoid the need for the `_isLowerCase` method.